server/controllers/items/lib/items.js

```diff
@@ -101,8 +101,12 @@ const items_ = module.exports = {
     items = items.map(item => Item.create(userId, item))
     const res = await db.bulk(items)
     const itemsIds = _.map(res, 'id')
-    const { docs } = await db.fetch(itemsIds)
-    await emit('user:inventory:update', userId)
+    const shelvesIds = _.deepCompact(_.map(items, 'shelves'))
```

Comment:
i would make it `_.uniq`

Reply:
you also need the `flatten` (array of arrays) and the compact (some items might not have an array of shelves), which is what `deepCompact` does in addition to `uniq` (maybe `deepCompact` could be renamed to highlight that)